wled00/udp.cpp

```diff
@@ -83,9 +88,16 @@ void realtimeLock(uint32_t timeoutMs, byte md)
 
   realtimeTimeout = millis() + timeoutMs;
   if (timeoutMs == 255001 || timeoutMs == 65000) realtimeTimeout = UINT32_MAX;
-  realtimeMode = md;
 
-  if (arlsForceMaxBri && !realtimeOverride) strip.setBrightness(scaledBri(255));
+  // Turn on if it was off
```

Comment:
This turns the strip on if it's off when realtime smoothing starts (it will turn off afterwards).

Reply:
Comparing this to the similar code in wled_serial.cpp (`if (!realtimeMode && bri == 0) strip.setBrightness(briLast);`) it may be unnecessary to scale briLast and check for realtime mode change.
Also it would be good to return state back to off once realtime ends if it was off when realtime started (I can't seem to find the part where your code does that).
Last thing, your code only executes arlsForceMaxBri override only on mode change whereas original did it always. Is this intentional?

Reply:
Thanks.  [Here](https://github.com/Aircoookie/WLED/blob/1f871690655a280c4a74bc681f7c0cb2d4d79247/wled00/udp.cpp#L211) is where the existing code _resets_ the strip to former brightness (using `bri` as the "saved value") once the UDP stream times out.  This also turns it _off_ if it was off beforehand, which is quite nice.  

For `arlsForceMaxBri`, I moved that inside the test for realtime mode turning on, since it seems resetting it to max brightness each and every time a packet arrives wouldn't be desired (or efficient).  For example, if the user decided it was too bright after realtime started, and slid the brightness slider down, you'd be surprised when it reset. But I'm not entirely clear how this variable is intended to be used, so could certainly go back to setting it back to max bright on every packet.

In terms of `scaledBri`, unless I'm mistaken, everywhere brightness is set this function is used so as to respect the maximum current limit and save your power supply.  So I suspect this is a bug in `wled_serial`.

Reply:
1. reviewing the code, I can se your point now since relay reacts to strip.getBrightness() and not bri as I initially thought.
2. arlsForceMaxBri (as it is most likely intended now) is unconditional override since it is a setting (as opposed to parameter)
3. you may have a point! :)